Task 5 NetworkClient

herrfreeman opened the pull request:
В клиент добавил 3 функции:
getVacancies - поиск вакансий
getCountries - получение справочника стран
getAreas - получает регион по id с вложенными/подчиненными регионами (можно по id страны искать)

herrfreeman committed:
task-5: Add network client with 3 functions: getVacancies, getCountries, getAreas, add test code of these ones to RootActivity

herrfreeman committed:
task-5: Fix detekt issues

IvanPodlesnykh commented on app/src/main/java/ru/practicum/android/diploma/data/network/AreasRequest.kt:
Почему реквест наследуется от респонз?

bukholovdmitry commented on app/src/main/java/ru/practicum/android/diploma/data/network/RetrofitNetworkClient.kt:
```diff
@@ -0,0 +1,105 @@
+    companion object {
+        private const val HTTP_NO_CONNECTION = -1
+        private const val HTTP_CLIENT_ERROR = 400
+        private const val HTTP_SUCCESS = 400
```
Здесь наверное когда запрос выполнился успешно HTTP_SUCCESS=200

herrfreeman committed:
task-5: Fix PR remarks

herrfreeman committed:
task-5: Merge from 'origin/develop'

# Conflicts:
#	app/src/main/java/ru/practicum/android/diploma/ui/root/RootActivity.kt

herrfreeman committed:
task-5: Fix more detekt remarks

herrfreeman committed:
task-5: Fix detekt (indentations)

lodrean commented on app/src/main/java/ru/practicum/android/diploma/data/network/Response.kt:
```diff
@@ -0,0 +1,5 @@
+package ru.practicum.android.diploma.data.network
+
+open class Response {
+    var resultCode = 0
```
можно создать константу DEFAULT_RESULT

lodrean commented on app/src/main/java/ru/practicum/android/diploma/data/network/CountriesRequest.kt:
```diff
@@ -0,0 +1,3 @@
+package ru.practicum.android.diploma.data.network
+
+class CountriesRequest
```
нужно добавить текст запроса
```suggestion
data class CountriesRequest(val expression: String)
```

herrfreeman committed:
task-5: Fix default result code
